src/core_test/mpas_test_openacc.F

```diff
@@ -0,0 +1,362 @@
+! Copyright (c) 2023 The University Corporation for Atmospheric Research (UCAR).
+!
+! Unless noted otherwise source code is licensed under the BSD license.
+! Additional copyright and license information can be found in the LICENSE file
+! distributed with this code, or at https://mpas-dev.github.io/license.html .
+!
+module mpas_test_core_openacc
+
+    use mpas_log, only : mpas_log_write
+
+    private
+
+    public :: mpas_test_openacc
+
+    contains
+
+    !-----------------------------------------------------------------------
+    !  function mpas_test_openacc
+    !
+    !> \brief Main driver for tests of OpenACC functionality in MPAS
+    !> \author G. Dylan Dickerson
+    !> \date   14 May 2024
+    !> \details
+    !>  This routine invokes tests for expected OpenACC behavior and any
+    !>  framework routines that are specific to OpenACC.
+    !>
+    !>  Return value: The total number of test that failed on any MPI rank.
+    !
+    !-----------------------------------------------------------------------
+    function mpas_test_openacc(domain) result(ierr_count)
+
+        use mpas_derived_types, only : domain_type
+        use mpas_kind_types, only : StrKIND
+        use mpas_dmpar, only : mpas_dmpar_max_int
+
+        implicit none
+
+        ! Arguments
+        type (domain_type), intent(inout) :: domain
+
+        ! Return value
+        integer :: ierr_count
+
+        ! Local variables
+        integer :: ierr, ierr_global
+        ! Use routine_name to track what is being tested next
+        character(len=StrKIND) :: routine_name
+
+        ierr_count = 0
+
+        call mpas_log_write('--- Begin OpenACC tests')
+
+        routine_name = 'openacc_test_rep_arrs'
```

Comment:
Since this name gets displayed in the log file with either `PASSED` or `FAILED`, should we choose a more descriptive string to print? Maybe something like "Simple CPU-GPU reproducibility"?

Reply:
I renamed the var to `test_log_str` and used your suggested string in the force push from [690909e ](https://github.com/MPAS-Dev/MPAS-Model/commit/690909ed0fb06b3037e997ed7828151e97f0ae2a)to [10fb55b ](https://github.com/MPAS-Dev/MPAS-Model/commit/10fb55bca892dec069413c0dca378ef2b21fcf12)